### PR TITLE
[SYCL] Assignment operator tests for 0-dim buffer accessor and local_accessor

### PR DESCRIPTION
**Pull request by andylshort**
SYCL2020 defines assignment operators for 0-dimensional accessors. This patch adds tests for the buffer `accessor` and `local_accessor` classes, with the `host_accessor` test being done as unit test in the intel/llvm repo instead.

### CODE REVIEW

**Review by dm-vodopyanov — APPROVED** [review]

**Comment by andylshort** [comment]
The implementation half of this has been merged as part of https://github.com/intel/llvm/pull/8069

File: SYCL/Basic/accessor/accessor.cpp [hunk 1]
```diff
@@ -1051,5 +1051,36 @@ int main() {
...
+  {
+    using LocalAccT = sycl::local_accessor<int, 0>;
+
+    sycl::queue Queue;
+
+    Queue.submit([&](sycl::handler &CGH) {
+      LocalAccT LocalAcc(CGH);
+      CGH.single_task<class local_acc_0_dim_assignment>([=]() {
+        LocalAcc = 64;
+        assert(LocalAcc == 64);
```
Comment (AlexeySachkov) [inline SYCL/Basic/accessor/accessor.cpp]: SYCL spec does not guarantee that `assert` will work in a kernel. Therefore, I would suggest to storing `LocalAcc` value into some other non-local accessor to check it on host

Comment (andylshort) [reply SYCL/Basic/accessor/accessor.cpp]: I should have now addressed this in https://github.com/intel/llvm-test-suite/pull/1570/commits/fe9cf992aa8236b5d4dfa8dc8790d0314f4bc928

**Comment by steffenlarsen** [comment]
Automatic CI is not running at the moment, so lets try verifying with the merged PR.

**Comment by AlexeySachkov** [comment]
> Automatic CI is not running at the moment, so lets try verifying with the merged PR.

Regular CI finished and it seems like the whole Linux run has failed due to some environment issue. However, `verify with` check passed, so I suggest that we merge this PR. @steffenlarsen, any objections?

**Comment by steffenlarsen** [comment]
The Linux failures should be addressed with next nightly build, but since verify is unaffected I agree with going ahead with a merge.

**Pull request merged**